server/events/project_command_runner.go

```diff
@@ -126,11 +126,22 @@ type ProjectOutputWrapper struct {
 func (p *ProjectOutputWrapper) Plan(ctx models.ProjectCommandContext) models.ProjectResult {
 	// Reset the buffer when running the plan. We only need to do this for plan,
 	// apply is a continuation of the same workflow
-	return p.updateProjectPRStatus(models.PlanCommand, ctx, p.ProjectCommandRunner.Plan)
+	result := p.updateProjectPRStatus(models.PlanCommand, ctx, p.ProjectCommandRunner.Plan)
+	p.updateOperationStatus(ctx, result)
+	return result
 }
 
 func (p *ProjectOutputWrapper) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
-	return p.updateProjectPRStatus(models.ApplyCommand, ctx, p.ProjectCommandRunner.Apply)
+	result := p.updateProjectPRStatus(models.ApplyCommand, ctx, p.ProjectCommandRunner.Apply)
+	p.updateOperationStatus(ctx, result)
+	return result
+}
+
+func (p *ProjectOutputWrapper) updateOperationStatus(ctx models.ProjectCommandContext, result models.ProjectResult) {
+	// No need to mark operation complete if failed operation.
+	if result.Failure == "" {
```

Comment:
What is the reason for this condition? Don't we want to close the channel if there is a failure? Customer will have to execute a new command to rerun the job. Which will generate a new job id.

Reply:
Yup, that makes sense! I don't think I need this check since we'd want to close any active receiver channels even if its a failed operation!

Reply:
Done!